### ISSUE TITLE
It is not possible to inject IEnumerable<CommandOption> using Microsoft.Extensions.DependencyInjection

### ISSUE THREAD

**Issue by mthamil** [natemcmaster/CommandLineUtils#85]
I am trying out the new dependency injection support for models. For `additionalServices`, I am passing an `IServiceProvider` created using `ServiceCollection.BuildServiceProvider()` in `Microsoft.Extensions.DependencyInjection`.

It seems that this service provider always returns an empty enumerable even if there are no implementations of a given service. I suppose this is by design due to the ability to retrieve all implementations of a given service. Since additional services are consulted before getting to the case where `IEnumerable<CommandOption>` is injected using `GetOptions()` in the custom CommandLineUtils service provider, these options can't be injected.

**Comment by natemcmaster** [comment]
If I understand right, you're trying to inject a constructor like this?
```
class MyCommand(IEnumerable<CommandOptions> options)
```

If so, yeah, I see the problem. It's here https://github.com/natemcmaster/CommandLineUtils/blob/59f11a6d01d399326a0f9b78bb9d7a826e3e13ef/src/CommandLineUtils/CommandLineApplication.cs#L925-L942. The addition services are consulted first. Given the `IEnumerable` behavior is not possible to disable in Microsoft.Extensions.DependencyInjection, I don't think it should be a problem to put `IEnumerable<CommandOptions>` first, before resolving additional services.

**Comment by mthamil** [comment]
Yes, that is exactly it. I didn't try it, but I assume it would happen with `IEnumerable<CommandArgument>` as well.

**Comment by natemcmaster** [comment]
This is an easy fix. I'm hoping to release 2.2.1 later this week, and this will be part of it.

**Comment by natemcmaster** [comment]
Fixed in 2.2.1 which is now on nuget.org

**Comment by mthamil** [comment]
Great, thanks!
